### PR TITLE
f-contact-preferences@0.2.0 - Initial component layout and styling

### PR DESCRIPTION
**Pull request by xander-marjoram**
### Added
- Initial component layout and styling

Still to-do (in later PRs):
- Testing
- Getting initial data from API
- Pushing updates to API

---

20211012T144238

### CODE REVIEW

**Review by oliversweb — DISMISSED** [review]

File: packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue [hunk 1]
```diff
@@ -26,21 +89,27 @@ export default {
...
-    border: 1px solid $color-red;
-    font-family: $font-family-base;
-    @include font-size(heading-m);
+    flex-flow: column;
+    border: none;
+    padding: 0;
+    margin: spacing(x2) 0;
 }
 
+.c-label-text--disabled {
```
Comment (dandel10n) [inline packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue]: We normally associate the class names with the component like `c-cardComponent-labelText--disabled` if they are written on the component level

File: packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue [hunk 2]
```diff
@@ -26,21 +89,27 @@ export default {
...
+    methods: {
+        onFormSubmit () {}
     }
 };
 </script>
 
 <style lang="scss" module>
-
-.c-contactPreferences {
+fieldset {
```
Comment (dandel10n) [inline packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue]: How does this work with the `module` hashing?

Comment (dandel10n) [reply packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue]: Wouldn't it better to add a class for this fieldset?

File: packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue (hunk 1 above)
Comment (xander-marjoram) [reply packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue]: Will update 👍🏻

File: packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue (hunk 2 above)
Comment (xander-marjoram) [reply packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue]: It does seem to work, but yes, I'll add a class

**Review by dandel10n — APPROVED** [review]

**Review by DamianMullins — APPROVED** [review]

File: packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue [hunk 3]
```diff
@@ -26,21 +89,27 @@ export default {
...
+    padding: 0;
+    margin: spacing(x2) 0;
 }
 
+.c-contactPreferences-labelText--disabled {
+    color: $color-content-disabled;
+}
+
+.c-contactPreferences-subtitle {
+    @include font-size(heading-s);
```
Comment (ashleynolan) [inline packages/components/organisms/f-contact-preferences/src/components/ContactPreferences.vue:113]: Extremely late PR comment – you can save yourself 3 lines of CSS by adding the `gamma` class to the tag instead (as we have [these helper classes as part of fozzie](https://github.com/justeat/fozzie/blob/v6.0.0-icing/src/scss/base/_typography.scss#L102-L113) that allow you to make different heading different sizes).

**Pull request merged**